[Resource] Deprecate Snapshot For Instance Snapshot (#5675)
IBM-Cloud/terraform-provider-ibm: website/docs/r/pi_instance_snapshot.html.markdown

---
subcategory: "Power Systems"
layout: "ibm"

> page_title: "IBM: pi_snapshot"

Reviewer: ```suggestion
page_title: "IBM: pi_instance_snapshot"
```

Author: updated

description: |-
  Manages instance snapshots in the Power Virtual Server cloud.
---

> # ibm_pi_snapshot

Reviewer: ```suggestion
# ibm_pi_instance_snapshot
```

> Creates, updates, deletes, and manages instance snapshots in the Power Virtual Server Cloud. For more information, about snapshots in the Power Virutal Server, see [snapshotting, cloning, and restoring](https://cloud.ibm.com/docs/power-iaas?topic=power-iaas-volume-snapshot-clone).

Reviewer: ```suggestion
Manages instance snapshots in the Power Virtual Server Cloud. For more information, about snapshots in the Power Virutal Server, see [snapshots for PVM Instance](https://cloud.ibm.com/apidocs/power-cloud#pcloud-pvminstances-snapshots-post).
```

Author: fixed

## Example usage

The following example enables you to create a snapshot:

```terraform
resource "ibm_pi_instance_snapshot" "testacc_snapshot"{
  pi_cloud_instance_id = "<value of the cloud_instance_id>"
  pi_description  = "Testing snapshot for instance"

>   pi_instance_name       = test-instance
>   pi_snap_shot_name       = test-snapshot

Reviewer: ```suggestion
  pi_instance_name       = "test-instance"
  pi_snap_shot_name       = "test-snapshot"
```

Author: updated

>   pi_snap_shot_name       = test-snapshot

Reviewer: ```suggestion
  pi_snapshot_name       = "test-snapshot"
```

Author: fixed

>   pi_volume_ids       = ["volumeid1","volumeid2"]

Reviewer: nit: format for readability.

Author: updated

}
```

### Notes

- Please find [supported Regions](https://cloud.ibm.com/apidocs/power-cloud#endpoint) for endpoints.
- If a Power cloud instance is provisioned at `lon04`, The provider level attributes should be as follows:
  - `region` - `lon`
  - `zone` - `lon04`

Example usage:

  ```terraform
    provider "ibm" {
      region    =   "lon"
      zone      =   "lon04"
    }
  ```
  
## Timeouts

The `ibm_pi_instance_snapshot` provides the following [Timeouts](https://www.terraform.io/docs/language/resources/syntax.html) configuration options:

- **create** - (Default 60 minutes) Used for Creating snapshot.
- **update** - (Default 60 minutes) Used for Updating snapshot.
- **delete** - (Default 10 minutes) Used for Deleting snapshot.

## Argument reference

Review the argument references that you can specify for your resource.

- `pi_cloud_instance_id` - (Required, String) The GUID of the service instance associated with an account.
- `pi_description` - (Optional, String) Description of the PVM instance snapshot.
- `pi_instance_name` - (Required, String) The name of the instance you want to take a snapshot of.

> - `pi_snap_shot_name` - (Required, String) The unique name of the snapshot.

Reviewer: ```suggestion
- `pi_snapshot_name` - (Required, String) The unique name of the snapshot.
```

Author: fixed

- If a Power cloud instance is provisioned at `lon04`, The provider level attributes should be as follows:
  - `region` - `lon`
  - `zone` - `lon04`

Example usage:

  ```terraform
    provider "ibm" {
      region    =   "lon"
      zone      =   "lon04"
    }
  ```
  
## Timeouts

The `ibm_pi_instance_snapshot` provides the following [Timeouts](https://www.terraform.io/docs/language/resources/syntax.html) configuration options:

- **create** - (Default 60 minutes) Used for Creating snapshot.
- **update** - (Default 60 minutes) Used for Updating snapshot.
- **delete** - (Default 10 minutes) Used for Deleting snapshot.

## Argument reference

Review the argument references that you can specify for your resource.

- `pi_cloud_instance_id` - (Required, String) The GUID of the service instance associated with an account.
- `pi_description` - (Optional, String) Description of the PVM instance snapshot.
- `pi_instance_name` - (Required, String) The name of the instance you want to take a snapshot of.
- `pi_snapshot_name` - (Required, String) The unique name of the snapshot.
- `pi_user_tags` - (Optional, List) The user tags attached to this resource.

> - `pi_volume_ids` - (Optional, String) A list of volume IDs of the instance that will be part of the snapshot. If none are provided, then all the volumes of the instance will be part of the snapshot.

Reviewer: Should not this be `List of strings`?
Also saw another instance below
```
`volume_snapshots` - (String) A map of volume snapshots included in the PVM instance snapshot.
```
How can a map be string type?

I am going to merge this PR, please correct in future PR.

Reviewer: @ismirlia @Alexander-Kita
